src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionBase.cs

```diff
@@ -146,7 +159,7 @@ protected void TraceConnection(Stream stream)
 
         public long GetLifetimeTicks(long nowTicks) => nowTicks - _creationTickCount;
```

Comment:
>  Can someone set super small connection lifetime?

Yes, we do not check granularity, it can be as low as 1ns.


> if it's now technically more correct to set _creationTickCount in MarkConnectionAsEstablished

Not sure about this. What I can see is that we'd need to change the field to not `readonly` and make sure we don't touch it in between ctor and `MarkConnectionAsEstablished` anywhere. So, what exact issue do you see with the current state? Is it something that would not happen if we move the setting up of `_creationTickCount` to `MarkConnectionAsEstablished`?